drivers/sdhc/Kconfig.imx

```diff
@@ -38,6 +38,14 @@ config IMX_USDHC_DMA_BUFFER_SIZE
 	help
 	  Size of USDHC ADMA descriptor buffer in bytes
 
+config IMX_USDHC_DMA_SCATTER_GATHER
+	bool "Scatter-Gather DMA mode"
+	depends on HEAP_MEM_POOL_SIZE > 0
+	select SDHC_SUPPORTS_SCATTER_GATHER_TRANSFER
+	help
+	  Enable Scatter-Gather DMA mode. Proper HEAP_MEM_POOL_SIZE should be
+	  configured for dynamic scatter gather list allocation.
+
```

Comment:
I was more thinking of eliminating this Kconfig option by a generic one too.

```
config IMX_USDHC
    select SDHC_SUPPORTS_SCATTER_GATHER_TRANSFER if HEAP_MEM_POOL_SIZE > 0
```

and then something like
```
config SDHC_SCATTER_GATHER_TRANSFER
    bool "Scatter-Gather API"
    depends on SDHC_SUPPORTS_SCATTER_GATHER_TRANSFER
```

and in the drivers `SDHC_SCATTER_GATHER_TRANSFER` would be used, here instead of `IMX_USDHC_DMA_SCATTER_GATHER`. This way activating Scatter-Gather would not need a driver specific Kconfig